server/src/main/java/org/opensearch/cluster/routing/OperationRouting.java

```diff
@@ -169,21 +189,37 @@ public GroupShardsIterator<ShardIterator> searchShards(
         final Set<IndexShardRoutingTable> shards = computeTargetedShards(clusterState, concreteIndices, routing);
         final Set<ShardIterator> set = new HashSet<>(shards.size());
         for (IndexShardRoutingTable shard : shards) {
-            ShardIterator iterator = preferenceActiveShardIterator(
-                shard,
-                clusterState.nodes().getLocalNodeId(),
-                clusterState.nodes(),
-                preference,
-                collectorService,
-                nodeCounts
-            );
+            ShardIterator iterator = null;
+            // TODO: Do we need similar changes in getShards call??
```

Comment:
searchShards is used in `_search_shards` API . getShards is used in search API . So putting it on only in `searchShards` doesn't completely serve our purpose. Is the right place for this is `shardRoutings` where even adaptive replica selection is also applied ?

Reply:
I think this makes sense. I had this in my mind as well, thats why had put a ```TODO``` line ```193```. I will move this call to ```shardRoutings```.